Dashboard/app/js/lib/views/selectors/device-selector.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+
+import Checkbox from 'akvo-flow/components/Checkbox';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/label-has-for */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+require('akvo-flow/views/react-component');
+
+FLOW.DeviceGroupSelectorView = FLOW.ReactComponentView.extend({
+  init() {
+    this._super();
+    this.deviceGroups = {};
+    this.deviceGroupNames = {};
+    this.handleChange = this.handleChange.bind(this);
+    this.deviceInAssignment = this.deviceInAssignment.bind(this);
+    this.renderDevices = this.renderDevices.bind(this);
+  },
+
+  didInsertElement(...args) {
+    this._super(...args);
+    if (FLOW.deviceGroupControl.content.isLoaded) {
+      FLOW.deviceGroupControl.get('content').forEach((item) => {
+        this.deviceGroupNames[item.get('keyId')] = item.get('code');
+        this.deviceGroups[item.get('keyId')] = {}; // initialize array of devices per group
+      });
+      if (FLOW.deviceControl.content.isLoaded) {
+        FLOW.selectedControl.selectedSurveyAssignment.get('devices').forEach((deviceId) => {
+          // populate pre-selected devices
+          const device = FLOW.Device.find(deviceId);
+          if (device && device.get('keyId')) {
+            FLOW.selectedControl.selectedDevices.pushObject(device);
+          }
+        });
+        FLOW.deviceControl.get('content').forEach((device) => {
+          this.deviceGroups[device.get('deviceGroup') ? device.get('deviceGroup') : 1][device.get('keyId')] = {
+            name: device.get('deviceIdentifier'),
+            checked: this.deviceInAssignment(device.get('keyId')),
+          };
+        });
+      }
+      this.renderDevices();
+    }
+  },
+
+  renderDevices() {
+    this.reactRender(
+      <div className="formSelectorList">
```

Comment:
shouldn't be called deviceSelectorList? or if it's common between the 2 then call it selectorList?